src/pymor/algorithms/rules.py

```diff
@@ -347,6 +349,11 @@ def apply_children(self, obj, children=None):
                 result[child] = {k: self.apply(v) if v is not None else v for k, v in c.items()}
             elif isinstance(c, (list, tuple)):
                 result[child] = tuple(self.apply(v) if v is not None else v for v in c)
+            elif isinstance(c, np.ndarray) and c.dtype == object and c.ndim >= 1:
```

Comment:
```suggestion
            elif isinstance(c, np.ndarray):
```

(Makes this a bit easier to read, I think. The checks are not needed here, as they have already been done by `get_children`.)